### PR TITLE
Set speedj_time from the parameter server

### PR DESCRIPTION
**Pull request by megantuttle**
When using the velocity-based controller, we were experiencing stuttering/jerky movements on our UR3.  After some investigating, I determined that this behavior was caused by the time parameter in the speedj() command, which was set too low for our setup (UR3 v.3.5, Kinetic, Ubuntu 16.04). I found that commands were not being received fast enough so stop commands were periodically issued, resulting in random stuttering/jerky movements.  The issue was fixed by increasing the time parameter, so the purpose of this PR is to add the speedj_time parameter to the parameter server so it can be easily adjusted (similar to the servoj_time parameter).

From the [UR scripting manual](https://www.universal-robots.com/download/) v3.3: 
> **speedj** (qd, a, t_min)
> Accelerate linearly in joint space and continue with constant joint
speed. The time t is optional; if provided the function will return after
time t, regardless of the target speed has been reached. If the time t is
not provided, the function will return when the target speed is reached
> 
> **Parameters**
> * qd: joint speeds [rad/s]
> * a: joint acceleration [rad/sˆ2] (of leading axis)
> * t_min: time [s] before the function returns (optional)

### CODE REVIEW

File: config/ur3_controllers.yaml [hunk 1]
```diff
@@ -64,26 +64,33 @@ vel_based_pos_traj_controller:
...
-      goal_time: 0.6
-      stopped_velocity_tolerance: 0.05
-      shoulder_pan_joint: {trajectory: 0.1, goal: 0.1}
-      shoulder_lift_joint: {trajectory: 0.1, goal: 0.1}
-      elbow_joint: {trajectory: 0.1, goal: 0.1}
-      wrist_1_joint: {trajectory: 0.1, goal: 0.1}
-      wrist_2_joint: {trajectory: 0.1, goal: 0.1}
-      wrist_3_joint: {trajectory: 0.1, goal: 0.1}
-   stop_trajectory_duration: 0.5
+      goal_time: 0.0
```
Comment (ipa-nhg) [inline config/ur3_controllers.yaml]: goal_time: 0.0 ???

make sense?

File: config/ur3_controllers.yaml [hunk 2]
```diff
@@ -64,26 +64,33 @@ vel_based_pos_traj_controller:
...
+      stopped_velocity_tolerance: 0.01
+      shoulder_pan_joint: {trajectory: 0.0, goal: 0.0}
+      shoulder_lift_joint: {trajectory: 0.0, goal: 0.0}
+      elbow_joint: {trajectory: 0.0, goal: 0.0}
+      wrist_1_joint: {trajectory: 0.0, goal: 0.0}
+      wrist_2_joint: {trajectory: 0.0, goal: 0.0}
+      wrist_3_joint: {trajectory: 0.0, goal: 0.0}
+   stop_trajectory_duration: 0.0
```
Comment (ipa-nhg) [inline config/ur3_controllers.yaml]: stop_trajectory_duration: 0.0 ???

make sense?

**Comment by miguelprada** [comment]
There are also quite a lot of changes unrelated to the main intention of this PR, i.e. adding a parameter to configure `speedj` time argument.

I don't see this being merged without a good cleanup.

**Comment by miguelprada** [comment]
The parameter wil only work for PolyScope version 3.3 and above. Would it be possible to make it so that if the `speedj_time` is set, then the provided value is used irrespective of the version, and the existing hardcoded versions are used otherwise?

I would also advise against setting default values in the launchfiles, since past experience has shown that the right value to use depends on the PolyScope version, and this can only be detected at runtime.

**Comment by megantuttle** [comment]
I was hesitant to apply the change to all PolyScope versions originally because I've only used v3.3 and up but I should have just looked at the other script manuals. I'll add it to the previous versions. Is there a reason that the driver doesn't set the time parameter for v3.1 and v3.2?

I see what you mean about being wary of setting default values in the launch files since it depends on the PolyScope version, but do we really want to make everyone set a speedj_time parameter? If the default is set to 20ms I don't think it would negatively affect anyone's current setup. If the default is set to 8ms it might cause some stuttering on the early versions, but even at top speed, the difference in travel distance between 20ms and 8ms would be very small.

**Comment by miguelprada** [comment]
The reason I'm hesitant to change the default behavior is because I've been bitten by adverse effects of setting this parameter badly in the past (i.e. #57, #100).

If we keep the default "hardcoded" values when no value is specified via a parameter, we should hopefully not break existing working instalations, while at the same time allowing anyone to override with whichever value works best for the version they're running.

That's why I suggest:
- if a value is specified using the parameter, use that value
- otherwise, keep existing "hardcoded" values based on the PolyScope version

**Comment by gavanderhoorn** [comment]
I'm going to close this as we've decided to deprecate `ur_modern_driver` due to the release of `ur_robot_driver`. Refer to the [announcement on ROS Discourse](https://discourse.ros.org/t/deprecation-of-ur-modern-driver/11629).

There were also some other aspects to this PR that were identified by @miguelprada.

Combined with the fact that these parameters are only needed for CB3 controllers, we'll not merge this in here.

@megantuttle: if you see a similar need in `ur_robot_driver`, please do open a PR there.

**Pull request closed without merging**